tests/ops/op_math/test_parametrized_hamiltonian.py

```diff
@@ -0,0 +1,370 @@
+        assert np.all(
+            [
+                qml.equal(op1, op2)
+                for op1, op2 in zip(test_example.H_ops_fixed, [qml.PauliX(0), qml.PauliY(1)])
+            ]
+        )
+
+    def test_H_parametrized_lists(self):
+        """Test that attributes H_parametrized_ops and H_parametrized_coeffs are as expected"""
+        assert test_example.H_coeffs_parametrized == [f1, f2]
+        assert np.all(
+            [
+                qml.equal(op1, op2)
+                for op1, op2 in zip(
+                    test_example.H_ops_parametrized, [qml.PauliZ(2), qml.Hadamard(3)]
+                )
+            ]
+        )
+
+    def test_H_fixed(self):
+        """Test that H_fixed is an Operator of the expected form"""
+        H_fixed = test_example.H_fixed()
+        op = qml.op_sum(qml.s_prod(1, qml.PauliX(0)), qml.s_prod(2, qml.PauliY(1)))
+        assert qml.equal(H_fixed, op)
+
+    def test_H_parametrized(self):
+        """Test H_parametrized is a function that, when passed parameters,
+        returns an Operator of the expected format."""
+        H_param = test_example.H_parametrized
+
+        H = H_param([1, 2], 0.5)
+        assert len(H) == 2
+        assert isinstance(H, qml.ops.Sum)
+        assert isinstance(H[0], qml.ops.SProd)
+        assert isinstance(H[1], qml.ops.SProd)
+
+    def test__repr__(self):
+        """Test repr method returns expected string"""
+        str = repr(test_example)
+        assert str == "ParametrizedHamiltonian: terms=4"
+
+    def test_wire_attribute(self):
+        """Tests that the wires attribute contains the expected wires, in the expected order"""
+        coeffs = [1, f1, 2, f2]
+        ops = [qml.PauliX(3), qml.PauliX(2), qml.PauliX(0), qml.PauliX(1)]
+
+        H = ParametrizedHamiltonian(coeffs, ops)
+        assert H.wires == H(param, 2).wires
+        assert H.wires == Wires([3, 0, 2, 1])
+
+
+class TestCall:
+
+    coeffs_and_ops_and_params = (
+        (
+            [f1, f2],
+            [qml.PauliX(0), qml.PauliY(1)],
+            [1, 2],
+        ),  # no H_fixed term, multiple H_parametrized terms
+        ([f1], [qml.PauliX(0)], [1]),  # no H_fixed term, one H_parametrized term
+        (
+            [1.2, 2.3],
+            [qml.PauliX(0), qml.PauliY(1)],
+            [],
+        ),  # no H_parametrized term, multiple H_fixed terms
+        ([1.2], [qml.PauliX(0)], []),  # no H_parametrized term, one H_fixed term
+        ([1, f1], [qml.PauliX(3), qml.PauliY(2)], [1]),  # one of each
+        ([1.2, f1, 2.3, f2], [qml.PauliX(0) for i in range(4)], [1, 2]),  # multiples of each
+    )
+
+    @pytest.mark.parametrize("coeffs, ops, params", coeffs_and_ops_and_params)
+    def test_call_succeeds_for_different_shapes(self, coeffs, ops, params):
+        """Test that calling the ParametrizedHamiltonian succeeds for
+        different numbers of H_fixed and H_parametrized terms"""
+        pH = ParametrizedHamiltonian(coeffs, ops)
+        pH(params, 0.5)
```

Comment:
should there not be an assert statement somewhere? Or is this just testing that no error is raised?

Reply:
I was just thinking about ensuring it didn't raise an error for some shapes (for example if the parametrized term was empty), but in hindsight I think it would make a lot of sense to at least also check that the number of terms is as expected in each case.